### PR TITLE
test(e2e): Add test for LDAP

### PR DESCRIPTION
**Pull request by moduli**
This PR adds an e2e test to cover LDAP functionality. 
- a new module was added to setup an LDAP docker container with a user and group
- a new enos scenario was created to use the module
- a new e2e test was added to go through an LDAP workflow.

The LDAP e2e test does the following:
- creates an LDAP auth method
- creates an LDAP account and attaches it to a boundary user
- confirms you can log in using the LDAP user's credentials
- creates an LDAP managed group
- attaches the LDAP managed group as a principal to a role

In addition, there was some refactoring done to reduce the number of enos scenarios. A new scenario was created (`e2e_docker_base_plus`) which will cover a boundary+target infra with other miscellaneous items (LDAP docker container, postgres credentials). The `key_destruction_test` was also moved to this scenario to reduce the overall run time of e2e tests (it's our longest test).

### CODE REVIEW

File: testing/internal/e2e/tests/base_plus/ldap_test.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+	"github.com/hashicorp/boundary/api/accounts"
+	"github.com/hashicorp/boundary/api/authmethods"
+	"github.com/hashicorp/boundary/api/managedgroups"
+	"github.com/hashicorp/boundary/testing/internal/e2e"
+	"github.com/hashicorp/boundary/testing/internal/e2e/boundary"
+	"github.com/stretchr/testify/require"
+)
+
+// TestCliLdap ...
```
Comment (moduli) [inline testing/internal/e2e/tests/base_plus/ldap_test.go:19]: Need to update this comment

File: testing/internal/e2e/tests/base_plus/ldap_test.go
Comment (moduli) [inline testing/internal/e2e/tests/base_plus/ldap_test.go:1]: @jimlambrt Here's an e2e test that does some LDAP operations. Looking to see if you think we need to add anything more here.

**Review by johanbrandhorst — APPROVED** [review]

File: enos/enos-scenario-e2e-docker-base-plus.hcl [hunk 2]
```diff
@@ -88,13 +88,24 @@ scenario "e2e_docker_base_with_postgres" {
     }
   }
 
+  step "create_ldap_server" {
+    module = module.docker_ldap
+    depends_on = [
+      step.create_docker_network
+    ]
+    variables {
+      image_name   = "${var.docker_mirror}/osixia/openldap:latest"
```
Comment (jimlambrt) [inline enos/enos-scenario-e2e-docker-base-plus.hcl:97]: that's an amazing docker img.

File: testing/internal/e2e/tests/base_plus/ldap_test.go
Comment (jimlambrt) [reply testing/internal/e2e/tests/base_plus/ldap_test.go:1]: other possibilities:

- make sure the ldap account is associated with the managed group
- make sure a user that's part of an ldap managed group gets the appropriate access for its roles

**Review by jimlambrt — COMMENTED** [review]
Looks great.  happy to discuss the other test cases.

File: testing/internal/e2e/tests/base_plus/ldap_test.go
Comment (moduli) [reply testing/internal/e2e/tests/base_plus/ldap_test.go:1]: Modified the test to include your suggestions.

**Pull request merged**